hail/python/hail/utils/struct.py

```diff
@@ -65,7 +65,7 @@ def __repr__(self):
         return str(self)
 
     def __str__(self):
-        return 'Struct({})'.format(', '.join('{}={}'.format(k, v) for k, v in self._fields.items()))
+        return 'Struct({})'.format(', '.join('{}={}'.format(k, repr(v)) for k, v in self._fields.items()))
```

Comment:
why repr inside str instead of str?

Reply:
has to do with strings, really. We moved to printing with quotes inside of `.show()`, and I think we really should be doing the same here.

Python collections print the repr inside of their elements:

```
>>> str(s)
"{'2', '1'}"
>>> repr(s)
"{'2', '1'}"
```

This avoids, for instance, printing escape chars inside strings. 

We still will print them if they're in a field name in the struct, but that's a problem for another day :)